### PR TITLE
Use actions@v3 and s0/git-publish-subdir-action to simplify workflow

### PR DESCRIPTION
**Pull request by s-k-z**
This change will write everything inside the `env.path` specified folder to the release branch without needing to manually copy individual files.
Use Github's builtin actions v3 just because it's new and cool.
Add workflow dispatch as an option to run (adds a button so you can manually start the workflow in the actions tab).
Prefer --immutable because --frozen-lockfile is deprecated.
No longer commits to release as O.A.F. (sorry!) but now you can see who actually made the commit that prompted the build.
Don't bother with the matrix builds, since KoLMafia doesn't run Nodejs, there's no need to support older versions.
Add a few comments so users can follow along/make changes easier.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -8,70 +8,71 @@ on:
...
+        run: yarn run build
+
+      - name: Push to Release
+        uses: s0/git-publish-subdir-action@develop
+        env:
+          REPO: self
+          BRANCH: release
+          FOLDER: ${{ env.path }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # GitHub will automatically add this - you don't need to bother getting a token
+          MESSAGE: "Build: ({sha}) {msg}" # The commit hash and message that prompted this workflow
```
Comment (spaghetti-squash) [inline .github/workflows/build.yml:78]: I think the commit hash is probably just going to clutter things up, but if you disagree, I am not particularly invested in that opinion.

Comment (s-k-z) [reply .github/workflows/build.yml:78]: https://github.com/loathers/greenbox/tree/release

Here's an example of what it looks like. It's a short hash so it's not too cluttered. Also it provides the link to the actual commit.

**Comment by s-k-z** [comment]
Some more changes I forgot that come with this: the KoLmafia folder is now used for your data. The build artifacts go into the same folder that the copied files would go which means building locally and the action are almost identical.

File: .github/workflows/build.yml (hunk 1 above)
Comment (spaghetti-squash) [reply .github/workflows/build.yml:78]: Oh it's the short version

Comment (spaghetti-squash) [reply .github/workflows/build.yml:78]: Yeah that' scool as hell

**Comment by spaghetti-squash** [comment]
I forgot this existed; is this good to merge and then migrate to loathers? or is the new plan just to remake this from scotch's template in loathers?

**Comment by s-k-z** [comment]
I think it's good to merge. Even if it doesn't end up in Loathers, maybe it can provide a good resource for people going forward.

**Pull request merged**